core/shared/src/main/scala/spire/math/Complex.scala

```diff
@@ -219,17 +219,21 @@ final case class Complex[@sp(Float, Double) T](real: T, imag: T)
     new Complex(t.log(abs), arg)
   }
 
+  // returns the root with angle in (-pi/2, pi/2]
   def sqrt(implicit f: Field[T], n0: NRoot[T], s: Signed[T]): Complex[T] = {
     if (isZero) {
       Complex.zero[T]
     } else {
       val two = f.fromInt(2)
-      val a = ((abs + real.abs) / two).sqrt
+      val a = ((abs + real) / two).sqrt
```

Comment:
Why should `a` be computed outside the match?

The whole function can be replaced by the following pseudo-code:

- `imag.isSignZero` and `real.isSignNegative`
  Return `Complex(f.zero, (-real).sqrt)`
- `image.isSignZero` and `real.isSignPositive` 
  Return `Complex(real.sqrt, f.zero)`
- Otherwise (add reference to [Wikipedia](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Negative_or_complex_square) )
  Compute and store `this.abs` in a temporary variable

The correction is fine.